src/app/services/currency.service.ts

```diff
@@ -0,0 +1,24 @@
+import { Injectable } from '@angular/core';
+import {Currency} from "../models/currency";
+import {of} from "rxjs";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class CurrencyService {
+getCurrency(){
+  const currency : Currency[] =[];
+  fetch('https://api.exchangerate.host/latest?base=ZAR')
+    .then((response)=> response.json())
+    .then((response)=> {
+      for (const [key, value] of Object.entries(response.rates)) {
```

Comment:
Why? Can you not just push `response.rates` into currencies?
`currency.push(...response.rates)`